### PR TITLE
Add restricted building rulesets

### PR DESCRIPTION
**Pull request by Nikolassparrow**
So something that I remembered from 2016 was the "restricted building" ruleset image on the server select screen that basically told you if a server allowed you to build in pois or not. I miss the hammer and triangle picture so this is something I've tried to work on for a while but finally managed to get working 😂 I found it from decompiling the CharacterSelect.gfx actionscript file, there were some other gamemodes included in there too!

So this PR: 

- Adds restricted building ruleset & image to server select screen
- Add isNoBuildInPois config variable for building in poi and removed allowPOIPlacement

### CODE REVIEW

File: data/2016/sampleData/defaultconfig.yaml [hunk 1]
```diff
@@ -102,7 +103,6 @@ speedtree:
   maxTreeHits: 20 # maximum hits it takes to chop a tree
 
 construction:
-  allowPOIPlacement: false
```
Comment (QuentinGruber) [inline data/2016/sampleData/defaultconfig.yaml:105]: maybe just comment it and add in the comment that isNoBuildInPois exist now

File: src/servers/ZoneServer2016/managers/constructionmanager.ts [hunk 2]
```diff
@@ -347,13 +346,14 @@ export class ConstructionManager {
...
   detectPOIPlacement(
+    server: ZoneServer2016,
     itemDefinitionId: number,
     position: Float32Array,
     client: Client,
     isInsidePermissionedFoundation: boolean
   ): boolean {
     if (client.isDebugMode) return false;
-    if (this.allowPOIPlacement) return false;
+    if (server.isNoBuildInPois) return false;
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/managers/constructionmanager.ts:356]: wouldbe better to do this directly when detectPOIPlacement method is called

**Review by QuentinGruber — CHANGES_REQUESTED** [review]

File: src/servers/ZoneServer2016/managers/constructionmanager.ts (hunk 2 above)
Comment (Nikolassparrow) [reply src/servers/ZoneServer2016/managers/constructionmanager.ts]: Done 👍

File: data/2016/sampleData/defaultconfig.yaml (hunk 1 above)
Comment (Nikolassparrow) [reply data/2016/sampleData/defaultconfig.yaml:105]: Done 👍

**Review by QuentinGruber — APPROVED** [review]

**Pull request merged**